homeassistant/components/http/ban.py

```diff
@@ -107,8 +107,10 @@ async def process_wrong_login(request):
 
     msg = f"Login attempt or request with invalid authentication from {remote_host} ({remote_addr})"
 
-    user_agent = request.headers.get("user-agent")
-    if user_agent:
+    user_agent_raw = request.headers.get("user-agent")
+    if user_agent_raw:
+        # Sanitize user agent to limited charset
+        user_agent = re.sub(r"[^a-zA-Z0-9:;,./() ]", "_", user_agent_raw)
```

Comment:
CommonMark treats _ as a special character that can be used as emphasis when it is the start and end of a word. This might lead to some odd looking results in the UI. Maybe space or entirely remove the offending characters?

Reply:
Good point. I'm against removing them silently in case any special characters were used as word boundaries.  I think a space is a good choice.